src/aap_eda/services/activation/engine/kubernetes.py

```diff
@@ -379,6 +393,15 @@ def _create_service(self, port: int) -> None:
                 LOGGER.info(f"Created Service: {service_name}")
             else:
                 LOGGER.info(f"Service already exists: {service_name}")
+                opened_ports = [
+                    port_item.port for port_item in service.items[0].spec.ports
+                ]
+                if bool(set(request_ports) - set(opened_ports)):
+                    raise ContainerStartError(
+                        f"Request ports {request_ports} is not opened in the "
+                        f"service {service_name} with ports: {opened_ports}"
+                    )
```

Comment:
If ports are plural, the verb should be "are", am I wrong? 
This is because the service already exists but the ports are different. Should we not recreate the service in that case?

Reply:
> If ports are plural, the verb should be "are", am I wrong? 
yes, changed.

> This is because the service already exists but the ports are different. Should we not recreate the service in that case?
We are not clear about the situation of the user's input service. It's not a good idea to recreate it . We can open the needed ports in the service, but it's hard to do the cleanup after activation is completed. The simple way is let user know and make the change.